File: frontend/src/constants/stages.js
// Canonical Stage Codes & Dynamic Arabic Mappings
export const STAGE_OPTIONS = [
  { code: 'ALL', label: 'جميع المراحل الخدمية ⛪' },
  { code: 'NURSERY', label: 'حضانة (KG1 & KG2)' },
  { code: 'PRIMARY_1', label: 'ابتدائي - الصف الأول' },
  { code: 'PRIMARY_2', label: 'ابتدائي - الصف الثاني' },
  { code: 'PRIMARY_3', label: 'ابتدائي - الصف الثالث' },
  { code: 'PRIMARY_4', label: 'ابتدائي - الصف الرابع' },
  { code: 'PRIMARY_5', label: 'ابتدائي - الصف الخامس' },
  { code: 'PRIMARY_6', label: 'ابتدائي - الصف السادس' },
  { code: 'PREPARATORY_1', label: 'إعدادي - الصف الأول' },
  { code: 'PREPARATORY_2', label: 'إعدادي - الصف الثاني' },
  { code: 'PREPARATORY_3', label: 'إعدادي - الصف الثالث' },
  { code: 'SECONDARY', label: 'ثانوي' },
  { code: 'UNIVERSITY', label: 'جامعة وخريجين' }
];

export const getStageLabel = (stageCodeOrName) => {
  if (!stageCodeOrName) return '';
  const match = STAGE_OPTIONS.find(
    s => s.code.toLowerCase() === stageCodeOrName.toLowerCase() || s.label === stageCodeOrName
  );
  return match ? match.label : stageCodeOrName;
};

File: backend/app/constants/stages.py
# Canonical Stage Codes and Arabic Translations
from typing import Dict, List, Any

CANONICAL_STAGES: Dict[str, str] = {
    "NURSERY": "حضانة (KG1 & KG2)",
    "PRIMARY_1": "ابتدائي - الصف الأول",
    "PRIMARY_2": "ابتدائي - الصف الثاني",
    "PRIMARY_3": "ابتدائي - الصف الثالث",
    "PRIMARY_4": "ابتدائي - الصف الرابع",
    "PRIMARY_5": "ابتدائي - الصف الخامس",
    "PRIMARY_6": "ابتدائي - الصف السادس",
    "PREPARATORY_1": "إعدادي - الصف الأول",
    "PREPARATORY_2": "إعدادي - الصف الثاني",
    "PREPARATORY_3": "إعدادي - الصف الثالث",
    "SECONDARY": "ثانوي",
    "UNIVERSITY": "جامعة وخريجين",
    "ALL": "جميع المراحل"
}


def get_stage_label(code: str) -> str:
    if not code:
        return ""
    return CANONICAL_STAGES.get(code.strip().upper(), code)


def get_all_stages() -> List[Dict[str, str]]:
    return [{"code": k, "label": v} for k, v in CANONICAL_STAGES.items()]
